include/revng/Pipes/TemporaryFile.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/ContainerFactory.h"
+#include "revng/Pipeline/ContainerSet.h"
+#include "revng/Pipeline/LLVMContainerFactory.h"
+#include "revng/Pipeline/Target.h"
+#include "revng/Support/Assert.h"
+
+namespace Pipeline {
+
+class TemporaryFile : public Container<TemporaryFile> {
```

Comment:
Document this and state that it can only contain a single target `{ "Root", K }` where `K` is passed at construction-time.